File: Problem List/14. Longest Common Prefix.py
from typing import List

# Best Big O
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]

        prefix = ''
        min_length = min(len(word) for word in strs)
        
        for i in range(min_length):
            for word in strs[1:]:
                if word[i] != strs[0][i]:
                    return prefix
            prefix += strs[0][i]

        return prefix

File: Problem List/14. Longest Common Prefix.c
char *longestCommonPrefix(char **strs, int strsSize)
{
  char *prefix = malloc(201);
  int i = 0;
  char letter;
  while (strs[0][i] != '\0')
  {
    letter = strs[0][i];
    for (int j = 1; j < strsSize; j++)
    {
      if (strs[j][i] == '\0' || strs[j][i] != letter)
      {
        prefix[i] = '\0';
        return prefix;
      }
    }
    prefix[i] = letter;
    i++;
  }
  prefix[i] = '\0';
  return prefix;
}
